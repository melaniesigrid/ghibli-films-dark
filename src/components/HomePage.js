import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import FilmCard from './FilmCard';
import Banner from '../assets/topImage.png';

function HomePage() {
  const [searchText, setSearchText] = useState('');
  const films = useSelector((state) => state.logic);
  const filteredFilms = films.filter((item) => Object.keys(item).some((key) => item[key]
    .toString()
    .toLowerCase()
    .includes(searchText.toLocaleLowerCase())));

  return (
    <div className="HomePage">
      <div className="BannerDiv">
        <img src={Banner} alt="Ghibli Logo" className="GhibliLogo" />
      </div>
      <div className="InputContainer">
        <input value={searchText} onChange={(e) => setSearchText(e.target.value)} type="text" placeholder="Search films..." className="SearchInput" />
      </div>
      <div className="FilmsContainer">
        {
          filteredFilms.map((film) => (
            <FilmCard
              key={film.filmKey}
              filmKey={film.filmKey}
              title={film.title}
              release={film.release}
              image={film.image}
            />
          ))
        }
      </div>
    </div>
  );
}

export default HomePage;
