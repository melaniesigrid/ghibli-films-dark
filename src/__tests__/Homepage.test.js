import React from 'react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import HomePage from '../components/HomePage';

const TestRenderer = require('react-test-renderer');

it('Renders homepage correctly', () => {
  const tree = TestRenderer
    .create(
      <Provider store={store}>
        <HomePage />
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
