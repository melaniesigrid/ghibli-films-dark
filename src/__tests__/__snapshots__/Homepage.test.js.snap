// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders homepage correctly 1`] = `
<div
  className="HomePage"
>
  <div
    className="BannerDiv"
  >
    <img
      alt="Ghibli Logo"
      className="GhibliLogo"
      src="topImage.png"
    />
  </div>
  <div
    className="FilmsContainer"
  />
</div>
`;
